.vsts-pipelines/templates/signalr-build.yml

```diff
@@ -67,4 +67,7 @@ jobs:
       buildConfiguration: $(BuildConfiguration)
       buildPlatform: $(AgentOsName)
 
+- ${{ if and(eq(parameters.runAsPublic, 'false'), ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```

Comment:
Why skip in these conditions? Wouldn't it make more sense to just not retain the build in that case? Why aren't we using retain-build.yml?

Reply:
Oh, didn't see this link: https://github.com/dotnet/arcade/blob/main/Documentation/SBOMGenerationGuidance.md#retention-rules-for-release-build-sboms
I'll look into that before merging

Copied the condition from arcade: https://github.com/dotnet/arcade/blob/78eaf78761027d225030be2b28aaf4e8bf392929/eng/common/templates/job/job.yml#L256-L257